core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -166,8 +167,26 @@ private[spark] object JettyUtils extends Logging {
     contextHandler
   }
 
+  private def addDefaultFilters(handlers: Seq[ServletContextHandler], conf: SparkConf): Unit = {
```

Comment:
Surely Jetty/netty already handle this internally? this is a hacky approach to supporting compression and I am all but certain modern containers (can be configured to) do this for you. Tomcat does.